### PR TITLE
use std::byte

### PR DESCRIPTION
**Pull request by neheb**
Signed-off-by: Rosen Penev <rosenp@gmail.com>

### CODE REVIEW

**Comment by KarlStraussberger** [comment]
Code scan throws

This pointer might have type `inotify_event` (size 16), but the pointer arithmetic here is done with type `byte *` (size 1).

**Comment by neheb** [comment]
Right. This PR changes nothing about that. I don't really know what the original code is doing.

File: src/util/jpeg_resolution.cc [hunk 1]
```diff
@@ -59,14 +61,14 @@ using uchar = unsigned char;
...
+static int get16m(std::byte* shrt)
 {
-    return (static_cast<uchar*>(const_cast<void*>(shrt))[0] << 8) | static_cast<uchar*>(const_cast<void*>(shrt))[1];
+    return std::to_integer<int>((shrt[0] << 8) | shrt[1]);
 }
 
 static int iohFgetc(const std::unique_ptr<IOHandler>& ioh)
 {
-    uchar c[1] = { 0 };
+    std::byte c[1] {};
```
Comment (KarlStraussberger) [inline src/util/jpeg_resolution.cc:71]: Does this init `c[0]` to 0 (see return statement)?

Comment (neheb) [reply src/util/jpeg_resolution.cc:71]: Should be identical. int() is equivalent to std::to_integer<int>

**Comment by neheb** [comment]
Removed first commit as that's being handled elsewhere.

**Pull request merged**